src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindSVBreakpointsSpark.java

```diff
@@ -0,0 +1,447 @@
+                    }
+                }
+                if ( op.consumesReferenceBases() )
+                    locus += ele.getLength();
+            }
+            return null;
+        }
+
+        private static boolean hasInitialSoftClip( final List<CigarElement> cigarElements, final GATKRead read ) {
+            final ListIterator<CigarElement> itr = cigarElements.listIterator();
+            if ( !itr.hasNext() ) return false;
+
+            CigarElement firstEle = itr.next();
+            if ( firstEle.getOperator() == CigarOperator.HARD_CLIP && itr.hasNext() ) {
+                firstEle = itr.next();
+            }
+            return firstEle.getOperator() == CigarOperator.SOFT_CLIP &&
+                    firstEle.getLength() >= MIN_SOFT_CLIP_LEN &&
+                    isHighQualityRegion(read.getBaseQualities(), 0);
+        }
+
+        private static boolean hasFinalSoftClip( final List<CigarElement> cigarElements, final GATKRead read ) {
+            final ListIterator<CigarElement> itr = cigarElements.listIterator(cigarElements.size());
+            if ( !itr.hasPrevious() ) return false;
+
+            CigarElement lastEle = itr.previous();
+            if ( lastEle.getOperator() == CigarOperator.HARD_CLIP && itr.hasPrevious() ) {
+                lastEle = itr.previous();
+            }
+            return lastEle.getOperator() == CigarOperator.SOFT_CLIP &&
+                    lastEle.getLength() >= MIN_SOFT_CLIP_LEN &&
+                    isHighQualityRegion(read.getBaseQualities(), read.getLength() - lastEle.getLength());
+        }
+
+        private static boolean isHighQualityRegion( final byte[] quals, int idx ) {
+            for ( final int end = idx+MIN_SOFT_CLIP_LEN; idx != end; ++idx ) {
+                if ( quals[idx] < MIN_QUALITY ) return false;
+            }
+            return true;
+        }
+    }
+
+    /**
+     * Class to detect reads that are part of a discordant pair.
+     * Concordant pairs are mapped to opposite strands of a reference sequence at a distance that is consistant with
+     * the library fragment length.
+     * Discordant pairs aren't concordant, except that we're also ignoring reads with mates mapped to other contigs.
+     * Class assumes that all reads presented to it are mapped.
+     */
+    @VisibleForTesting static final class DiscordantPairDetector {
+        private static final int FUNKY_PAIR_LOCUS_WIDTH = 100; // uncertainty in event locus for discordant pair
+        //TODO: Above should depend on the fragment size distribution
+
+        public EventLocus getLocusIfDiscordant( final GATKRead read, final long readId ) {
+            if ( read.mateIsUnmapped() ) {
+                final int locus = read.isReverseStrand() ? read.getStart() - FUNKY_PAIR_LOCUS_WIDTH : read.getEnd();
+                return new EventLocus(locus, FUNKY_PAIR_LOCUS_WIDTH, readId);
+            } else if ( read.getContig() == read.getMateContig() ) {
```

Comment:
`GATKRead` and its super-interface `Locatable` do not make any promises regarding interning of contig names (or other Strings) in their method contracts. It's true that htsjdk's `SAMRecord` happens to use interning in a few places, but we shouldn't be relying on that in GATK.